src/axom/inlet/Table.hpp

```diff
@@ -1000,45 +1014,46 @@ class Table : public Verifiable
 
   /*!
    *****************************************************************************
-   * \brief Get an array represented as an unordered map from the input file
+   * \brief Get a container represented as an unordered map from the input file
    * of primitive type
    *
-   * \param [out] map Unordered map to be populated with array contents
+   * \param [out] map Unordered map to be populated with container contents, will be
+   * cleared before contents are added
    *
-   * \return Whether or not the array was found
+   * \return Whether or not the container was found
    *****************************************************************************
    */
-  template <typename T>
+  template <typename Key, typename T>
   typename std::enable_if<detail::is_inlet_primitive<T>::value, bool>::type
-  getArray(std::unordered_map<int, T>& map) const
+  getContainer(std::unordered_map<Key, T>& map) const
   {
     map.clear();
-    if(!axom::utilities::string::endsWith(m_name, CONTAINER_GROUP_NAME))
+    if(!isContainerGroup(m_name))
     {
       return false;
     }
-    for(auto& item : m_fieldChildren)
+    for(const auto& item : m_fieldChildren)
     {
-      auto pos = item.first.find_last_of("/");
-      int index = std::stoi(item.first.substr(pos + 1));
+      auto index = detail::toIndex<Key>(removeBeforeDelimiter(item.first));
       map[index] = item.second->get<T>();
     }
     return true;
   }
 
   /*!
    *****************************************************************************
-   * \brief Get an array represented as an unordered map from the input file
+   * \brief Get a container represented as an unordered map from the input file
    * of user-defined type
    *
-   * \param [out] map Unordered map to be populated with array contents
+   * \param [out] map Unordered map to be populated with container contents, will be
+   * cleared before contents are added
    *
-   * \return Whether or not the array was found
+   * \return Whether or not the container was found
    *****************************************************************************
    */
-  template <typename T>
+  template <typename Key, typename T>
```

Comment:
Minor: When I saw a `Key` type for a map, I expected its counterpart to be something like `Val`, rather than `T`.

Reply:
I'll go ahead and make sure all these templates are `<Key, Val>` - I think I might use `<K, V>` in a few places as well.

Interestingly enough, the [standard](https://eel.is/c++draft/unord.map.syn#header:%3cunordered_map%3e) uses `<Key, T>` for associative containers.